core/whistle_services-1.0.0/src/whs_account_conversion.erl

```diff
@@ -0,0 +1,174 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2012-2014, 2600Hz, INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%-------------------------------------------------------------------
+-module(whs_account_conversion).
+
+-export([make_reseller/1
+         ,force_make_reseller/1
+        ]).
+-export([demote_reseller/1
+         ,force_demote_reseller/1
+        ]).
+-export([cascade_reseller_id/2]).
+-export([set_reseller_id/2]).
+-export([has_reseller_descendants/1]).
+
+-include("whistle_services.hrl").
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Set the reseller status on the provided account and update all
+%% sub accounts
+%% @end
+%%--------------------------------------------------------------------
+-spec make_reseller(ne_binary()) -> {'error', _} | 'ok'.
+make_reseller(Account) ->
+    AccountId = wh_util:format_account_id(Account, 'raw'),
+    case whapps_util:is_master_account(AccountId) of
+        'true' -> {'error', 'master_account'};
+        'false' ->
+            case has_reseller_descendants(AccountId) of
+                'true' -> {'error', 'reseller_descendants'};
+                'false' -> do_make_reseller(AccountId)
+            end
+    end.
+
+-spec force_make_reseller(ne_binary()) -> {'error', _} | 'ok'.
+force_make_reseller(Account) ->
+    AccountId = wh_util:format_account_id(Account, 'raw'),
+    do_make_reseller(AccountId).
+
+-spec do_make_reseller(ne_binary()) -> {'error', _} | 'ok'.
+do_make_reseller(AccountId) ->
+    _ = update_account_definition(AccountId, <<"pvt_reseller">>, 'true'),
+    _ = maybe_update_services(AccountId, <<"pvt_reseller">>, 'true'),
+    io:format("promoting account ~s to reseller status, updating sub accounts~n", [AccountId]),
```

Comment:
Since these functions can be called from Crossbar's `cb_accounts`, `lager` should be use instead of `io`